SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1684,33 +1684,73 @@ USpatialActorChannel* ActorSystem::TryRestoreActorChannelForStablyNamedActor(AAc
 
 void ActorSystem::InvokeRepNotifies()
 {
-	for (FObjectRepNotifies& ObjectRepNotifies : RepNotifiesToSend)
+	// Call repnotifies on actors before subobjects
+	TArray<FObjectRepNotifies> Actors;
+	TArray<FObjectRepNotifies> Subobjects;
+	for (auto ObjectToRepNotifiesIt = RepNotifiesToSend.CreateIterator(); ObjectToRepNotifiesIt; ++ObjectToRepNotifiesIt)
 	{
-		UObject* Object = ObjectRepNotifies.Object.Get();
-		if (!IsValid(Object))
+		FObjectRepNotifies& ObjectRepNotifies = ObjectToRepNotifiesIt->Value;
+		if (ObjectRepNotifies.RepNotifies.Num() == 0)
 		{
 			continue;
 		}
-		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Object);
-		if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+
+		UObject* Object = ObjectToRepNotifiesIt->Key.Get();
+		ObjectRepNotifies.Object = Object;
+
+		if (Cast<AActor>(Object))
 		{
-			// TODO UNR-5785 - Once fixed, change log level to warning
-			UE_LOG(LogActorSystem, Log, TEXT("Failed to invoke rep notifies for an object as its entity id was invalid. Object: %s"),
-				   *Object->GetName());
-			continue;
+			Actors.Add(MoveTemp(ObjectRepNotifies));
 		}
-		USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId);
-		if (!IsValid(Channel))
+		else
 		{
-			UE_LOG(LogActorSystem, Warning,
-				   TEXT("Failed to invoke rep notifies for an object as its channel was invalid. Object: %s, Entity: %lld"),
-				   *Object->GetName(), EntityId);
-			continue;
+			Subobjects.Add(MoveTemp(ObjectRepNotifies));
 		}
-		RemoveRepNotifiesWithUnresolvedObjs(*Object, *Channel, ObjectRepNotifies.RepNotifies);
-		Channel->PostReceiveSpatialUpdate(Object, ObjectRepNotifies.RepNotifies, ObjectRepNotifies.PropertySpanIds);
 	}
-	RepNotifiesToSend.Empty();
+
+	for (FObjectRepNotifies& ActorRepNotifies : Actors)
+	{
+		TryInvokeRepNotifiesForObject(ActorRepNotifies);
+	}
+	for (FObjectRepNotifies& ObjectRepNotifies : Subobjects)
+	{
+		TryInvokeRepNotifiesForObject(ObjectRepNotifies);
+	}
+
+	RepNotifiesToSend.Empty(floor(RepNotifiesToSend.Num() * 0.9));
```

Comment:
Use 'FMath::Floor' rather than 'floor', and try to avoid magic numbers - though personally I would Empty() unless the profiler suggests it is a problem.

Reply:
I think you're probably right about Empty(), changed 😄